manifests/instance/tls.pp

```diff
@@ -0,0 +1,167 @@
+# @summary Configure TLS for an instance
+#
+# Requires LDAPI to be enabled
+#
+# @author https://github.com/simp/pupmod-simp-ds389/graphs/contributors
+#
+define ds389::instance::tls (
+  String[2]                                 $root_dn,
+  Stdlib::Absolutepath                      $root_pw_file,
+  Variant[Boolean, Enum['disabled','simp']] $ensure        = simplib::lookup('simp_options::pki', { 'default_value' => false}),
+  Simplib::Port                             $port          = 636,
+  String[1]                                 $source        = simplib::lookup('simp_options::pki', { 'default_value' => "/etc/pki/simp/${module_name}_${title}"}),
+  Stdlib::Absolutepath                      $cert          = "/etc/pki/simp_apps/${module_name}_${title}/x509/public/${facts['fqdn']}.pub",
+  Stdlib::Absolutepath                      $key           = "/etc/pki/simp_apps/${module_name}_${title}/x509/private/${facts['fqdn']}.pem",
+  Stdlib::Absolutepath                      $cafile        = "/etc/pki/simp_apps/${module_name}_${title}/x509/cacerts/cacerts.pem",
+  Ds389::ConfigItems                        $dse_config    = simplib::dlookup('ds389::instance::pki', 'dse_config', { 'default_value' => {} }),
+  String[16]                                $token         = simplib::passgen("ds389_${title}_pki", { 'length' => 32 }),
+  String[1]                                 $service_group = 'dirsrv'
+) {
+  assert_private()
+
+  $_instance_name = split($title, /^(dirsrv@)?slapd-/)[-1]
```

Comment:
ds389::instance does not allow the title to begin with dirsrv@ or slapd-, so why do we allow it here?

Reply:
It was an old safety check, I'll just remove it.